packages/react-scripts/scripts/utils/createJestConfig.js

```diff
@@ -54,6 +55,12 @@ module.exports = (resolve, rootDir, isEjecting) => {
     'coverageReporters',
     'coverageThreshold',
     'snapshotSerializers',
+    // These are custom, we should be able to change these if required:
+    'automock',
+    'coveragePathIgnorePatterns',
+    'globals',
+    'noStackTrace',
+    'testPathIgnorePatterns',
```

Comment:
The default Jest configuration for create-react-app is pretty limiting.  Here are the reasons for enabling each key:

**`automock`**
The default is `false`, but user should be able to utilize all mocks without explicit declaration

**`coveragePathIgnorePatterns`**
The user should be able to specify directories or files to ignore rather than have to use `istanbul-ignore` comments all over the codebase

**`globals`**
This is helpful in preventing errors in tests (e.g. mocking the `window` object)

**`testPathIgnorePatterns`**
Helpful for ignoring helper or mock files in test directories.